src/routes/MyPage.jsx

```diff
@@ -0,0 +1,288 @@
+import { useEffect, useState } from "react";
+import { Link } from "react-router-dom";
+import { SmallPost } from "../components/Posts";
+import {
+  getPosts,
+  getTags,
+  getUserInMyPage,
+  updateUserInMyPage,
+} from "../apis/api";
+import { getCookie } from "../utils/cookie";
+
+const MyPage = () => {
+  const [postList, setPostList] = useState([]);
+  const [profile, setProfile] = useState({
+    id: "",
+    user: {},
+    college: "",
+    major: "",
+  });
+  useEffect(() => {
+    const getPostsAPI = async () => {
+      const posts = await getPosts();
+      setPostList(posts);
+    };
+    getPostsAPI();
+  }, []);
+  const [isEditEmail, setIsEditEmail] = useState(false);
+  const [isEditUsername, setIsEditUsername] = useState(false);
+  const [isEditCollege, setIsEditCollege] = useState(false);
+  const [isEditMajor, setIsEditMajor] = useState(false);
```

Comment:
이부분의 경우, 굳이 state를 4개나 작성할 필요가 없기 때문에
```suggestion
  const [isEdit, setIsEdit] = useState({
   email : false,
   username : false,
   college : false,
   major : false,
  });
```

이런식으로 작성하는게 더 효율적일 것 같습니당